WordPress/gutenberg
# Add/mobile fullscreen image preview

## Description
This PR covers fullscreen image preview in the gutenberg mobile editor as specified in [gb-mobile issue 1286](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1286)

WP iOS PR for testing: https://github.com/wordpress-mobile/WordPress-iOS/pull/12834
![fullscreenimage-sketch](https://user-images.githubusercontent.com/1103838/63370328-cc9cf700-c336-11e9-9b63-bd19709f1ef8.png)
Remaining tasks:
1. [x] Fade animation to fullscreen image preview from Image block editor
2. [x] Image Preview has correct aspect ratio in portrait and landscape
3. [x] Highlight border around image when image block is selected (same as video block)

Decided to commit behind a `__DEV__` flag until rotation issues are resolved. Tasks below will be in a future PR. 
_4.  Address dependency on Fix for Android modal rotation crash ([wp-mobile PR 10383](https://github.com/wordpress-mobile/WordPress-Android/pull/10383#issuecomment-521123642))
5.  (•••) Button over Image
6.  (•••) Button leads to BottomSheet with Edit and ViewFullscreen options_



## How has this been tested?



Tested as a submodule to gutenberg-mobile repo at latest develop ([cf27fba](https://github.com/wordpress-mobile/gutenberg-mobile/commit/cf27fba4ce3a8243450ee52b0909a3508a6efd4a) )
### Test steps:
1. Open article in the block editor.
2. Navigate to an image block with an image loaded, or add a new image block and add an image
3. Select the image block and notice the blue border shows around the image
4. Press the image while the image block is selected and notice the image will be viewed in fullscreen.
5. Press outside the boundaries of the image in fullscreen mode, or swipe the image down to exit the fullscreen image preview. 

*Note you can also test rotating the device while in fullscreen preview, but there is a known issue that will cause the image preview to close on rotate when testing in WordPress Android on a phone running API 28 - See WordPress Android issues [10227](https://github.com/wordpress-mobile/WordPress-Android/issues/10227) and [1333](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1333)

## Screenshots 
Fade in to fullscreen image preview:
![fullscreen-preview-v0](https://user-images.githubusercontent.com/1103838/63369719-682d6800-c335-11e9-80db-b9dfb5b6fd94.gif)


## Types of changes




New feature - this code updates the image block of the mobile editor so that users can do a fullscreen preview of the image that they are adding. 

## Checklist:
